src/AStar/Generalised.elm

```diff
@@ -98,9 +99,11 @@ updateCost current neighbour model =
             Dict.insert neighbour distanceTo model.costs
 
         distanceTo =
-            reconstructPath newCameFrom neighbour
-                |> Array.length
-                |> toFloat
+            neighbourCostFn current neighbour
+                + (reconstructPath newCameFrom neighbour
+                    |> Array.length
+                    |> toFloat
+                  )
```

Comment:
In the end this is the place where the neighbour cost function fits in, not the one I guessed in the issue #9.